File: frontend/src/models/moneyunit.ts
import { Member, buildMember } from "./member";

export interface MoneyUnit {
  owner: Member;
  value: number;
}

export const buildMoneyUnit = ({
  owner = buildMember({}),
  value = 0
}: Partial<MoneyUnit>): MoneyUnit => ({
  owner,
  value,
});

File: backend/src/main/java/com/github/wcosmedlr/dto/MoneyUnit.java
package com.github.wcosmedlr.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Data
@SuperBuilder(setterPrefix = "set")
@NoArgsConstructor
public class MoneyUnit implements Comparable<MoneyUnit> {

    private Member owner;
    private Double value;

    @Override
    public int compareTo(MoneyUnit o) {
        return this.getValue().compareTo(o.getValue());
    }

}
